### PR TITLE
create auth. services

### PR DESCRIPTION
**Pull request by Pgarza-dev**
created auth functions for reg users & login users

### CODE REVIEW

File: js/services/auth.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const postData = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(userData),
+    };
+    const response = await fetch(registerUrl, postData);
+    const json = await response.json();
+    console.log(json);
```
Comment (kyrregjerstad) [inline js/services/auth.js:25]: We don't want the console logs in the codebase, they are only needed for us when we develop

File: js/services/auth.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+}
+
+const userToRegister = {
+  name: "PabloGarza",
+  email: "pabgar91211@stud.noroff.no",
+  password: "thisismypassword",
+};
+
+registerUser(userToRegister);
```
Comment (kyrregjerstad) [inline js/services/auth.js:37]: do we want to call the function here?

File: js/services/auth.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    const json = await response.json();
+    console.log(json);
+    const accessToken = json.accessToken;
+    localStorage.setItem("accessToken", accessToken);
+  } catch (error) {
+    console.log(error + "Something went wrong");
+  }
+}
+
+const userToLogin = {
```
Comment (kyrregjerstad) [inline js/services/auth.js:63]: we should probably not commit you username and email to the codebase :P

**Review by kyrregjerstad — CHANGES_REQUESTED** [review]
Most of it looks great, let's make some of the small changes and then merge it 👍

File: js/services/auth.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import { API_BASE_URL } from "../consts.js";
+const registerUrl = `${API_BASE_URL}/social/auth/register`;
+
+// 1. Register User. Create a function that takes in a url and a user object
```
[the remark this answers is not in the document]
Comment (Pgarza-dev) [reply js/services/auth.js:4]: done!

File: js/services/auth.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * API call to register a user
+ * @param {string} url
+ * @param {register object} userData
+ * ```js
+ * registerUser( registerUrl, userToRegister);
+ */
+async function registerUser(userData) {
```
[the remark this answers is not in the document]
Comment (Pgarza-dev) [reply js/services/auth.js:13]: done!

File: js/services/auth.js (hunk 3 above)
Comment (Pgarza-dev) [reply js/services/auth.js:63]: done!

File: js/services/auth.js (hunk 1 above)
Comment (Pgarza-dev) [reply js/services/auth.js:25]: done!

File: js/services/auth.js (hunk 2 above)
Comment (Pgarza-dev) [reply js/services/auth.js:37]: done!

**Review by kyrregjerstad — APPROVED** [review]
looks great! 🚀

**Pull request merged**